src/newgrf_config.cpp

```diff
@@ -711,9 +709,9 @@ GRFConfig *GetGRFConfig(uint32_t grfid, uint32_t mask)
 std::string GRFBuildParamList(const GRFConfig *c)
 {
 	std::string result;
-	for (uint i = 0; i < c->num_params; i++) {
+	for (const uint32_t &value : c->param) {
 		if (!result.empty()) result += ' ';
-		result += std::to_string(c->param[i]);
+		result += std::to_string(value);
 	}
```

Comment:
ranges got anything for this?

Reply:
Not as far as I know, it's a bit specific.